Documentation/Plugins.md

```diff
@@ -0,0 +1,220 @@
+
+#### Declaring a build tool plugin in the package manifest
+
+Like all kinds of plugins, build tool plugins are declared in the package manifest.  There is a `pluginTarget` entry in the `targets` section of the package, and if the plugin should be visible to other packages, there needs to be a corresponding entry in the `products` section as well:
+
+```swift
+// swift-tools-version: 5.6
+import PackageDescription
+
+let package = Package(
+    name: "MyPluginPackage",
+    products: [
+        .plugin(
+            name: "MyBuildToolPlugin",
+            targets: [
+                "MyBuildToolPlugin"
+            ]
+        )
+    ],
+    dependencies: [
+        .package(
+            url: "https://github.com/example/sometool",
+            from: "0.1.0"
+        )
+    ],
+    targets: [
+        .plugin(
+            name: "MyBuildToolPlugin",
+            capability: .buildTool(),
+            dependencies: [
+                .product(name: "SomeTool", package: "sometool"),
+            ]
+        )
+    ]
+)
+```
+
+The `plugin` target declares the name and capabilities of the plugin, along with its dependencies.  The capability of `.buildTool()` is what indicates this as a build tool plugin as opposed to any other kind of plugin — this also determines what entry point the plugin is expected to implement (see below).
+
+The  `plugin` product is what makes the plugin visible to other packages that have dependencies on the package that defines the plugin.  The name of the plugin doesn't have to match the name of the product, but they are often the same in order to avoid confusion.  If a built tool plugin is used only within the package that declares it, there is no need to declare a `plugin` product.
+
+#### Build tool target dependencies
+
+The dependencies specify the command line tools that will be available for use in commands constructed by the plugin.  Each dependency can be either a `executableTarget` or a `binaryTarget` target in the same package, or can be an `executable` product in another package (there are no binary products).  In the example above, the plugin depends on the hypothetical _SomeTool_ product in the _sometool_ package on which the package that defines the plugin has a dependency.  Note that this does not necessarily mean that _SomeTool_ will have been built when the plugin is invoked — it only means that the plugin will be able to look up the path at which the tool will exist when any commands constructed by the plugin are run.
+
+Executable dependencies are built for the host before the build starts, while binary dependencies are references to `artifactbundle` archives that contains prebuilt binaries (see [SE-305](https://github.com/apple/swift-evolution/blob/main/proposals/0305-swiftpm-binary-target-improvements.md)).  Binary targets are often used when the tool is built using a different build system than SwiftPM or when building it on use is prohibitively expensive or requires a special build environment.
```

Comment:
> Executable dependencies are built for the host before the build starts

Is that the current state or the desired design? My understanding was that we did SE-305 because this isn't yet the case, but I might be wrong.

Reply:
Speaking of Xcode targeting iOS (since I have not yet had a chance to try it with raw SwiftPM such as Linux to Wasm), the status quo in Xcode 13.3 seems to be that executables are built for the host *by the time they are needed* (not necessarily at the very beginning), but that they are also built for the target platform. In the iOS case, that leads to compilation failure since executables are not supported, which in turn makes such build tool plugins completely unusable for cross‐compilation (at least local ones; maybe a conditional dependency would work for an executable from another package?).

Reply:
I updated this now and hopefully the new wording is more correct.